### PR TITLE
oembed providers

### LINKED ISSUE #2356: Switch from "wontfix" core patch to oembed_providers module
## Issue
hook_oembed_providers_alter() is going to be marked as #wontfix by core.

Patch from:
https://www.drupal.org/project/drupal/issues/3008119

## Proposed Solution

- [x] Need to switch to https://www.drupal.org/project/oembed_providers which now has the alter as of 1.1.0
- [x] Remove patch from composer.json
- [x] Remove/replace sitenow_media_wysiwyg_oembed_providers_alter() and sitenow_media_wysiwyg_media_source_info_alter()
- [x] Make sure no functionality is lost related to youtube, vimeo, kaltura, twitter, facebook, instagram

### LINKED ISSUE #2667: Consider removing oEmbed patch related to #2656
# Issue

Relates to https://github.com/uiowa/uiowa/issues/2656

Google has supposedly fixed this issue: https://issuetracker.google.com/issues/174576256#comment28

# Proposed Solution

- Consider removing the patch introduced in #2656 
- Test to make sure we aren't seeing the same error. Alex encountered it while trying to add a new YouTube video to a website.
- Wait for core to implement a more robust fallback.

### LINKED ISSUE #2720: YouTube oEmbed videos with `www` links will not embed correctly.
# Problem
Recently youtube changed some things that made `www` links for OEmbed not embed correctly. This has the potential for breaking all videos that are currently embedded with `www` links. 

# Steps to reproduce
Try to embed any youtube video with the www link in the URL and it will not work, saying that it `Could not retrieve the oEmbed resource.`

# Proposed solution
Possibly run an update hook on all video assets to replace all  instances of `https://www.youtube.com/watch?v=` with `https://youtu.be/`

### PR DESCRIPTION
**Pull request by joewhitsitt**
Resolves https://github.com/uiowa/uiowa/issues/2356
Resolves https://github.com/uiowa/uiowa/issues/2667
Resolves https://github.com/uiowa/uiowa/issues/2720

Cleaning this up now as Youtube fixed the original issue. Drupal 9.2 has a more robust fallback to prevent future issues with youtube and other providers related to https://github.com/uiowa/uiowa/issues/2667, https://github.com/uiowa/uiowa/issues/2720, so I would advise moving to 9.2 sooner than later to prevent future disruption upstream.

# Stories

As a site editor I am still able to view existing remote video embeds and add new Kalutra Mediaspace public videos from mediaspace.uiowa.edu using the oembed share url.

As a webmaster I cannot access oembed providers configuration.

As a developer, my site syncs correctly and it is no longer dependent on a core patch/hooks.

As a site editor, I can throw any supported youtube link (public) at the remote video functionality and have it work.

# To Test

- sync a site with existing remote video embeds (e.g. sandbox.uiowa.edu)
- note that oembed providers module is enabled and config is imported.
- webmasters can't access the oembed providers configuration
- existing youtube and kaltura videos are still rendering
- you can add a youtube and kaltura video (e.g. https://mediaspace.uiowa.edu/id/1_hhz13skb)
- try various youtube urls schemes: https://www.youtube.com, https://youtube.com, https://youtu.be - you should not get `Could not retrieve the oEmbed resource`

### CODE REVIEW

**Comment by joewhitsitt** [comment]
Nevermind. Something I want to possibly add to this.

File: config/default/oembed_providers.provider.uiowa_kaltura.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+langcode: en
+status: true
+dependencies: {  }
+id: uiowa_kaltura
+label: 'University of Iowa'
+provider_url: 'http://mediaspace.uiowa.edu'
+endpoints:
+  -
+    schemes:
+      - 'http://mediaspace.uiowa.edu/*'
```
Comment (richardbporter) [inline config/default/oembed_providers.provider.uiowa_kaltura.yml:11]: Is it necessary to allow http scheme here?

**Comment by richardbporter** [comment]
The format `youtu.be/` fails with the `?watch` query string param. Is that expected? It works without it.

Fails: https://youtu.be/watch?v=cvh0nX08nRw
Works: https://youtu.be/cvh0nX08nRw

File: config/default/oembed_providers.provider.uiowa_kaltura.yml (hunk 1 above)
Comment (joewhitsitt) [reply config/default/oembed_providers.provider.uiowa_kaltura.yml:11]: not sure. i was just following some documentation way back and copied over from what we currently have.

**Comment by joewhitsitt** [comment]
can you tell me how you get these variations? clicking a video on their website uses the `https://www.youtube.com/watch?v=...` format while the share button on the video gives me `https://youtu.be/...` without the watch.

File: config/default/oembed_providers.provider.uiowa_kaltura.yml (hunk 1 above)
Comment (richardbporter) [reply config/default/oembed_providers.provider.uiowa_kaltura.yml:11]: Ok. I just assumed that browsers would block mixed content from being embedded. I tried http and it gets embedded with a different https URL so I'm not sure this is an issue. Allowing http feels odd though.

**Comment by richardbporter** [comment]
Yea I think the first one isn't going to happen unless someone manipulates the URL, which is what I did. I think that is expected to fail.

**Comment by joewhitsitt** [comment]
I want to say not to worry about those kind of manipulations, but I also see that scheme in https://oembed.com/providers.json

File: config/default/oembed_providers.provider.uiowa_kaltura.yml (hunk 1 above)
Comment (richardbporter) [reply config/default/oembed_providers.provider.uiowa_kaltura.yml:11]: I guess since it gets rewritten as https, I'm ok with this.

**Review by richardbporter — APPROVED** [review]
Just tried https://youtu.be/watch?v=cvh0nX08nRw  in prod now and it doesn't work, so no change there.

**Pull request merged**